src/main/java/seedu/address/logic/commands/taskcommands/UpdateTaskCommand.java

```diff
@@ -0,0 +1,223 @@
+package seedu.address.logic.commands.taskcommands;
+
+import java.util.Date;
+
+import javafx.collections.ObservableList;
+import seedu.address.commons.collections.UniqueItemCollection.DuplicateItemException;
+import seedu.address.commons.collections.UniqueItemCollection.ItemNotFoundException;
+import seedu.address.commons.core.Messages;
+import seedu.address.commons.exceptions.IllegalValueException;
+import seedu.address.commons.util.DateUtil;
+import seedu.address.logic.commands.CommandResult;
+import seedu.address.model.task.DeadlineTask;
+import seedu.address.model.task.Description;
+import seedu.address.model.task.EventTask;
+import seedu.address.model.task.FloatingTask;
+import seedu.address.model.task.Task;
+
+/**
+ * Updates a task identified using it's last displayed index from TaskManager.
+ */
+public class UpdateTaskCommand extends TaskCommand {
+
+	public static final String COMMAND_WORD = "update";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Updates the task identified by the index number used in the last task listing.\n"
+            + "Depending on whether 'task', 'description' or 'date' is stated, the task will be updated accordingly.\n"
+            + "1) Parameters: INDEX (must be a positive integer) task UPDATED_VALUE\n"
+            + "Example: " + COMMAND_WORD + " 1 task Meeting from Oct 31 to Nov 1\n"
+            + "2) Parameters: INDEX (must be a positive integer) description UPDATED_VALUE\n"
+            + "Example: " + COMMAND_WORD + " 1 description Meeting in town\n"
+            + "3) Parameters: INDEX (must be a positive integer) date UPDATED_VALUE\n"
+            + "Example: " + COMMAND_WORD + " 1 date Oct 31 to Nov 1";
+
+    public static final String MESSAGE_UPDATE_TASK_SUCCESS = "Updated task: %1$s";
+    public static final String MESSAGE_CANNOT_UPDATE_TASK = "Selected task's description cannot be updated";
+    public static final String MESSAGE_DEADLINE_TASK_REQUIRED = "Deadline task required for update of deadline";
+    public static final String MESSAGE_EVENT_TASK_REQUIRED = "Event task required for update of start date and end date";
```

Comment:
"Deadline task required for update of deadline" and "Event task required for update of start date and end date" - I think it's hard for users to understand this?